File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/style/CursorAnimator.java

package io.github.abc15018045126.sora.widget.style;

/**
 * Interface for provide various cursor animations
 *
 * @author abc15018045126, Dmitry Rubtsov
 */
public interface CursorAnimator {

    /**
     * Mark the current cursor position as animation start position
     */
    void markStartPos();

    /**
     * Mark the current cursor position as animation end position
     */
    void markEndPos();

    /**
     * Start animation
     */
    void start();

    /**
     * Cancel animation
     */
    void cancel();

    /**
     * Check whether animation is in process
     */
    boolean isRunning();

    /**
     * The current x position of cursor in view offset
     */
    float animatedX();

    /**
     * The current y position of cursor in view offset
     */
    float animatedY();

    /**
     * Height of current line background
     */
    float animatedLineHeight();

    /**
     * Bottom Y position in view offset of current line background
     */
    float animatedLineBottom();
}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/style/CursorAnimator.kt
package io.github.abc15018045126.sora.widget.style

/**
 * Interface for provide various cursor animations
 *
 * @author abc15018045126, Dmitry Rubtsov
 */
interface CursorAnimator {

    /**
     * Mark the current cursor position as animation start position
     */
    fun markStartPos()

    /**
     * Mark the current cursor position as animation end position
     */
    fun markEndPos()

    /**
     * Start animation
     */
    fun start()

    /**
     * Cancel animation
     */
    fun cancel()

    /**
     * Check whether animation is in process
     */
    fun isRunning(): Boolean

    /**
     * The current x position of cursor in view offset
     */
    fun animatedX(): Float

    /**
     * The current y position of cursor in view offset
     */
    fun animatedY(): Float

    /**
     * Height of current line background
     */
    fun animatedLineHeight(): Float

    /**
     * Bottom Y position in view offset of current line background
     */
    fun animatedLineBottom(): Float
}
